src/main/java/org/repodriller/persistence/csv/CSVFile.java

```diff
@@ -16,95 +16,157 @@
 	}
 
+	public CSVFile (String fileName, String[] header) {
+		this(fileName, false, header);
+	}
+
 	public CSVFile(String path, String name, boolean append) {
-		this(verifyPath(path) + name, append);
-	}		
-	
-	public CSVFile (String path, String name , String [] header) {
-		this(verifyPath(path) + name, header);
+		this(Paths.get(path, name).toAbsolutePath().toString(), append, null);
 	}
-	
-	public CSVFile (String fileName, String [] header) {
-		this(fileName,header,false);
+
+	public CSVFile (String path, String name, String[] header) {
+		this(Paths.get(path, name).toAbsolutePath().toString(), false, header);
 	}
 
-	public CSVFile(String fileName, String[] header, boolean append ){
-		this.header = header;
-		try {
-			ps = new PrintStream(new FileOutputStream(fileName, append));
-			if (header != null)	{
-				String headers = Arrays.toString(header);
-				headers = headers.substring(1, headers.length()-1);
-				ps.println(headers);	
-			}
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
+	public CSVFile(String path, String name, boolean append, String[] header){
+		this(Paths.get(path, name).toAbsolutePath().toString(), append, header);
 	}
-	
+
+	@Deprecated
+	public CSVFile(String fileName, String[] header, boolean append){
+		this(fileName, append, header);
+	}
+
+	/* Interface PersistenceMechanism. */
+
+	/**
+	 * Write a CSV record consisting of these fields.
+	 *
+	 * @arg fields	Objects with toString()'s. As many fields as are in the header, if you provided one
+	 */
 	@Override
-	public synchronized void write(Object... line) throws CSVFileFormatException {
-		if (header != null && header.length != line.length)
-			throw new CSVFileFormatException("CSV Header Columns Number Differs From Writer Columns Number.");
-		
-		boolean first = true;
-		for(Object o : line) {
-			if(!first) ps.print(",");
-
-			if(o == null) ps.print("null");
+	public synchronized void write(Object... fields) throws CSVFileFormatException {
+		if (!writable)
+			throw new RepoDrillerException("Error, writing to a closed CSVFile");
+
+		if (header != null && header.length != fields.length)
+			throw new CSVFileFormatException("CSV header had " + header.length + " columns but you provided " + fields.length + " fields");
+
+		/* Convert fields to CSV-friendly strings. */
+		String[] escapedFields = Stream.of(fields).map(field -> {
+			if (field == null)
+				return "null";
 			else {
-				String field = o.toString();
-				field = StringEscapeUtils.escapeCsv(field);
-				ps.print(field);
+				String raw = field.toString();
+				if (raw == null)
+					return "null";
+				return StringEscapeUtils.escapeCsv(raw);
 			}
-			first=false;
-		}
-		
-		ps.println();
-		ps.flush();
+		})
+		.toArray(String[]::new);
+
+		/* Out it goes. */
+		String csvRecord = String.join(",", escapedFields);
+		log.debug("Writing csvRecord: " + csvRecord);
+		ps.println(csvRecord);
 
+		ps.flush(); /* TODO This has performance implications. */
```

Comment:
I think this dates back when repodriller crashed when an exception happens. Now that repodriller never crashes (awesome!! kk), we may leave it to the stream to decide when to flush. What do you think?

Reply:
Discussion in #92